src/NHapi.Base/Parser/XMLParser.cs

```diff
@@ -25,796 +25,830 @@
+
+                            var segmentConstructTypes = new Type[] { typeof(IMessage) };
+                            var segmentConstructArgs = new object[] { null };
+                            var s = (ISegment)reps[j].GetType().GetConstructor(segmentConstructTypes).Invoke(segmentConstructArgs);
+                            xp.Parse(s, root);
+                            var doc2 = new XmlDocument();
+                            var root2 = doc2.CreateElement(s.GetType().FullName);
+                            doc2.AppendChild(root2);
+                            xp.Encode(s, root2);
+                            var out2 = new StringWriter();
+                            var ser = XmlWriter.Create(out2);
+                            doc.WriteTo(ser);
+                            if (out2.ToString().Equals(out_Renamed.ToString()))
+                            {
+                                Console.Out.WriteLine("Re-encode OK");
+                            }
+                            else
+                            {
+                                Console.Out.WriteLine("Warning: XML different after parse and re-encode: \r\n" + out2.ToString());
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                SupportClass.WriteStackTrace(e, Console.Error);
+            }
+        }
+
+        /// <summary> Returns a String representing the encoding of the given message, if
+        /// the encoding is recognized.  For example if the given message appears
+        /// to be encoded using HL7 2.x XML rules then "XML" would be returned.
+        /// If the encoding is not recognized then null is returned.  That this
+        /// method returns a specific encoding does not guarantee that the
+        /// message is correctly encoded (e.g. well formed XML) - just that
+        /// it is not encoded using any other encoding than the one returned.
+        /// Returns null if the encoding is not recognized.
+        /// </summary>
+        public override string GetEncoding(string message)
+        {
+            string encoding = null;
+
+            // check for a number of expected strings
+            var expected = new string[] { "<MSH.1", "<MSH.2", "</MSH>" };
+            var isXML = true;
+            for (var i = 0; i < expected.Length; i++)
+            {
+                if (message.IndexOf(expected[i]) < 0)
+                {
+                    isXML = false;
+                }
+            }
+
+            if (isXML)
+            {
+                encoding = "XML";
+            }
+
+            return encoding;
+        }
+
+        /// <summary> Returns true if and only if the given encoding is supported
+        /// by this Parser.
+        /// </summary>
+        public override bool SupportsEncoding(string encoding)
+        {
+            if (encoding.Equals("XML"))

```

Comment:
```csharp
returns encoding.Equals("XML");
```

alot simpler to return the result of the expression 😁

Reply:
done